src/components/TooltipNew/GuideTooltip.js

```diff
@@ -0,0 +1,272 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './style.scss';
+import PopperTooltip from './PopperTooltip';
+import TooltipPortal from './TooltipPortal';
+
+const baseClass = 'guide-tooltip';
+
+const cx = classNames.bind(styles);
+
+const addPadding = (rect, padding) => {
+  const x = Math.round(rect.x) - padding;
+  const y = Math.round(rect.y) - padding;
+  const width = Math.round(rect.width) + 2 * padding;
+  const height = Math.round(rect.height) + 2 * padding;
+  const top = y;
+  const left = x;
+  const bottom = top + height;
+  const right = left + width;
+
+  return {
+    x,
+    y,
+    width,
+    height,
+    top,
+    left,
+    bottom,
+    right
+  };
+};
+
+class VirtualReference {
+  constructor(element) {
+    this.element = element;
+  }
+
+  getBoundingClientRect() {
+    return addPadding(this.element.getBoundingClientRect(), 8);
+  }
+
+  get clientWidth() {
+    return this.getBoundingClientRect().width;
+  }
+
+  get clientHeight() {
+    return this.getBoundingClientRect().height;
+  }
+}
+
+const GapOverlay = ({ gap, isVisible, slide }) => {
```

Comment:
move it to its own file

Reply:
shouldn't it be names OverlayGap ?

Reply:
The way I see it is that an `OverlayGap` would be a gap itself created for an overlay :)

This is actually an overlay with a gap. I changed the name to `SpotlightOverlay` - better ?